// @mui/x-internal-gestures/matchers/mocks/MockBadUpdateOptionsGesture.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MockBadUpdateOptionsGesture = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _core = require("../../core");
class MockBadUpdateOptionsGesture extends _core.Gesture {
  constructor(...args) {
    super(...args);
    this.state = {};
    this.isSinglePhase = void 0;
    this.eventType = void 0;
    this.optionsType = void 0;
    this.mutableOptionsType = void 0;
    this.mutableStateType = void 0;
  }
  // eslint-disable-next-line class-methods-use-this
  resetState() {}
  clone(overrides) {
    return new MockBadUpdateOptionsGesture((0, _extends2.default)({
      name: this.name
    }, overrides));
  }

  // We remove the updateOptions implementation
  // eslint-disable-next-line class-methods-use-this
  updateOptions() {}
}
exports.MockBadUpdateOptionsGesture = MockBadUpdateOptionsGesture;
